### PR TITLE
NeutronInducedTrap fix - solver parameter assertion

### PR DESCRIPTION
**Pull request by jhdark**
## Proposed changes

Solver parameters for NeutronInducedTrap traps could not assert new values for solver parameters from the defaults defined in ExtrinsicTrapBase. A simple example of this is shown below,  in which an ExtrinsicTrap is able to update the absolute tolerance from 1.0 to 2, whereas the NeutronInducedTrap does not.

Additionally a small typo in the doc strings is updated

```python
import festim as F

trap_extrinsic = F.ExtrinsicTrap(
    1,
    0,
    1,
    0,
    "mat_name",
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    absolute_tolerance=2,
    relative_tolerance=3,
    maximum_iterations=4,
    linear_solver="example",
)
trap_neutron = F.NeutronInducedTrap(
    1,
    0,
    1,
    0,
    "mat_name",
    1,
    1,
    1,
    1,
    0,
    absolute_tolerance=5,
    relative_tolerance=6,
    maximum_iterations=7,
    linear_solver="example_2",
)
print(trap_extrinsic.absolute_tolerance)
print(trap_neutron.absolute_tolerance)

```

## Types of changes

by simply adding a
```python 
**kwargs
```
into the 
```python3
super().__init__()
```
for the NeutronInducedTrap, one can access the solver parameters

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [x] Documentation Update (if none of the other choices apply)
- [x] New tests

## Checklist

- [x] Black formatted
- [x] Unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: festim/concentration/traps/neutron_induced_trap.py [hunk 1]
```diff
@@ -7,7 +7,7 @@ class NeutronInducedTrap(ExtrinsicTrapBase):
     Class for neutron induced trap creation with annealing.
     The temporal evolution of the trap density is given by
 
-    dn_t/dt = phi*K*(1 - n_t/n_max) + A_0*exp(-E_A/(k_B*T))*n_t
+    dn_t/dt = phi*K*(1 - n_t/n_max) - A_0*exp(-E_A/(k_B*T))*n_t
```
Comment (RemDelaporteMathurin) [inline festim/concentration/traps/neutron_induced_trap.py:10]: How is this related to the bug?

Comment (jhdark) [reply festim/concentration/traps/neutron_induced_trap.py:10]: Its not, just a small typo that needed fixing anyway

Comment (RemDelaporteMathurin) [reply festim/concentration/traps/neutron_induced_trap.py:10]: Mb I thought that was the formulation itself. Shouldn't review PRs on my phone.....

File: test/unit/test_traps/test_neutron_induced_trap.py [hunk 2]
```diff
@@ -124,3 +124,38 @@ def test_create_form_density(self):
         print(expected_form)
         print(self.my_trap.form_density)
         assert self.my_trap.form_density.equals(expected_form)
+
+
+class TestNeutronInducedTrapSolverParameters:
```
[the remark this answers is not in the document]
Comment (jhdark) [reply test/unit/test_traps/test_neutron_induced_trap.py:129]: Not sure what you mean here

[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply test/unit/test_traps/test_neutron_induced_trap.py:129]: Here you have a test that checks that the attributes are correct when the values are passsed at instanciation.

You should also check they can be changer after instanciation.

**Comment by jhdark** [comment]
Not sure why the codecov test is failing on these recent commits

File: test/unit/test_traps/test_neutron_induced_trap.py [hunk 3]
```diff
@@ -124,3 +124,49 @@ def test_create_form_density(self):
...
+        assert self.my_trap.maximum_iterations == 13
+        assert self.my_trap.linear_solver == "mumps"
+
+    def test_attributes_change_since_instanciation(self):
+        """
+        Test to ensure values can be updated from an original
+        instantation
+        """
+
+        expected_form = 3.6
```
Comment (RemDelaporteMathurin) [inline test/unit/test_traps/test_neutron_induced_trap.py:169]:
```suggestion
        expected_tolerance= 3.6
```

File: test/unit/test_traps/test_neutron_induced_trap.py [hunk 4]
```diff
@@ -124,3 +124,49 @@ def test_create_form_density(self):
...
+        self.my_trap.absolute_tolerance = 3.6
+
+        assert self.my_trap.absolute_tolerance == expected_form
```
Comment (RemDelaporteMathurin) [inline test/unit/test_traps/test_neutron_induced_trap.py:172]:
```suggestion
        assert self.my_trap.absolute_tolerance == expected_tolerance
```

File: test/unit/test_traps/test_neutron_induced_trap.py [hunk 5]
```diff
@@ -124,3 +124,49 @@ def test_create_form_density(self):
...
+
+        assert self.my_trap.absolute_tolerance == 2.5
+        assert self.my_trap.relative_tolerance == 1.2e-10
+        assert self.my_trap.maximum_iterations == 13
+        assert self.my_trap.linear_solver == "mumps"
+
+    def test_attributes_change_since_instanciation(self):
+        """
+        Test to ensure values can be updated from an original
+        instantation
```
Comment (RemDelaporteMathurin) [inline test/unit/test_traps/test_neutron_induced_trap.py:166]:
```suggestion
        Test to ensure values can be updated after instanciation
```

**Comment by RemDelaporteMathurin** [comment]
> Not sure why the codecov test is failing on these recent commits

Funny enough, it seems it's the `__init__.py` file that got impacted... I wouldn't worry too much about it

https://app.codecov.io/gh/RemDelaporteMathurin/FESTIM/pull/534

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
